packages/connect/src/base/stun.ts

```diff
@@ -368,3 +371,69 @@ function generateRequests(multiAddrs: Multiaddr[]): Request[] {
     tId: stun.generateTransactionId()
   }))
 }
+
+/**
+ * Tries to determine the external IPv4 address
+ * @returns Addr+Port or undefined if the STUN response are ambiguous (e.g. bidirectional NAT)
+ *
+ * @param multiAddrs Multiaddrs to use as STUN servers
+ * @param socket Node.JS socket to use for the STUN request
+ * @param runningLocally set to true when running a local testnet
+ */
+export async function getExternalIp(
+  multiAddrs: Multiaddr[] | undefined,
+  socket: Socket,
+  runningLocally = false
+): Promise<ConnectionInfo | undefined> {
+  let responses: RequestWithResponse[] = []
+  if (runningLocally) {
+    if (multiAddrs == undefined || multiAddrs.length == 0) {
+      const socketAddress = socket.address()
+      log(
+        `Running in local-mode without any given STUN server, assuming that socket address 127.0.0.1:${socketAddress.port} is public address`
+      )
+      return {
+        ...socketAddress,
+        address: '127.0.0.1'
+      }
+    }
+
+    responses.push(...(await iterateThroughStunServers(multiAddrs, socket, Infinity, true)))
+
+    if (responses.length == 0) {
+      log(`Cannot determine external IP because running in local mode and none of the local STUN servers replied.`)
+      return
+    }
+  } else {
+    if (multiAddrs == undefined || multiAddrs.length == 0) {
+      responses.push(...(await iterateThroughStunServers(PUBLIC_STUN_SERVERS, socket)))
+    } else {
+      responses.push(...(await iterateThroughStunServers(multiAddrs, socket)))
+
+      // We need at least two answers to determine whether the node
+      // operates behind a bidirectional NAT
+      if (responses.length < 2) {
+        responses.push(...(await iterateThroughStunServers(PUBLIC_STUN_SERVERS, socket)))
+      }
+    }
+  }
+
+  if (responses.length < 2) {
+    // We have tried all ways to check if the node
+    // is operating behind a bidirectional NAT but we
+    // could not get more than one response from STUN servers
+    log(
+      `Could not get more than one response from available STUN servers. Assuming that node operates behind a bidirectional NAT`
+    )
+    return
+  }
+
+  const interpreted = intepreteResults(responses)
+
+  if (interpreted.ambiguous) {
+    log(`Received STUN results are ambiguous. Assuming that node operates behind a bidirectional NAT`)
+    return
+  } else {
+    return interpreted.publicAddress
+  }
```

Comment:
```suggestion
  if (!interpreted.ambiguous) {
    return interpreted.publicAddress
  }
  log(`Received STUN results are ambiguous. Assuming that node operates behind a bidirectional NAT`)
```

Reply:
Would keep it as is to prevent from checking if `publicAddress == undefined` or sth like that